src/interfaces/external/IBalancerPool.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+interface IBalancerPool {
+    function getMainToken() external view returns (address);
+
+    function getRate() external view returns (uint256);
+
+    function getInvariant() external view returns (uint256);
+
+    function getLastInvariant() external view returns (uint256);
+
+    function getFinalTokens() external view returns (address[] memory);
+
+    function getNormalizedWeight(address token) external view returns (uint256);
+
+    function getNormalizedWeights() external view returns (uint256[] memory);
+
+    function getSwapFee() external view returns (uint256);
+
+    function getNumTokens() external view returns (uint256);
+
+    function getBalance(address token) external view returns (uint256);
+
+    function totalSupply() external view returns (uint256);
+
+    function getPoolId() external view returns (bytes32);
+
+    function decimals() external view returns (uint8);
+
+    function joinPool(uint256 poolAmountOut, uint256[] calldata maxAmountsIn) external;
+
+    function swapExactAmountIn(
+        address tokenIn,
+        uint256 tokenAmountIn,
+        address tokenOut,
+        uint256 minAmountOut,
+        uint256 maxPrice
+    ) external returns (uint256 tokenAmountOut, uint256 spotPriceAfter);
+
+    function swapExactAmountOut(
+        address tokenIn,
+        uint256 maxAmountIn,
+        address tokenOut,
+        uint256 tokenAmountOut,
+        uint256 maxPrice
+    ) external returns (uint256 tokenAmountIn, uint256 spotPriceAfter);
+
+    function joinswapExternAmountIn(
```

Comment:
These function names are what the actual mainnet contracts call them, so we can not change their name, otherwise our calls would revert.

I dont think we will use this interface function, but if we ended up using it, and we changed the function name to camel case, then our calls would revert because the real mainnet contract function is called `joinswapExternalAmountIn ` not `joinSwapExternalAmountIn `.